### PR TITLE
Add CI to update vyper info list

### PR DESCRIPTION
**Pull request by kuksag**
- Successful [run](https://github.com/kuksag/solc-bin/runs/7877584142?check_suite_focus=true) on [branch](https://github.com/kuksag/solc-bin/tree/kuksag/vyper-compiler-ci-test) 
- Autocommit that came after the job: https://github.com/kuksag/solc-bin/commit/2867bccbdabbd327d39fdce4e2a93a954de3cc8d , that pushed `vyper.list.json` to the branch 

We may also want to edit the message and author of the commit to distinguish the CI actions 

Alas, there is no option to run this job on every Vyper release, since we don't have rights for Vyper repo and we can't create webhook events: https://stackoverflow.com/questions/58465057/trigger-a-github-action-when-another-repository-creates-a-new-release

### CODE REVIEW

File: .github/workflows/build_vyper_list.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Prerequisites
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y curl wget jq findutils coreutils sed
```
Comment (kuksag) [inline .github/workflows/build_vyper_list.yaml:18]: I'm not sure whether `findutils` and `coreutils` are installed by default, so I add it anyway

File: .github/workflows/build_vyper_list.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Cache downloads
+        id: cache-primes
+        uses: actions/cache@v3
+        with:
+          path: vyper-bin
+          key: ${{ runner.os }}-vyper-bin
```
Comment (kuksag) [inline .github/workflows/build_vyper_list.yaml:25]: This helps us to cache downloaded binaries

File: .github/workflows/build_vyper_list.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: build-vyper-list
+
+on:
+  push:
+  schedule:
+    # every day at midnight
+    - cron: "0 0 * * *"
```
Comment (leviska) [inline .github/workflows/build_vyper_list.yaml:7]: it's better to run it several times per day. make it run every 3 hours

File: vyper_fetch.sh [hunk 4]
```diff
@@ -1,10 +1,16 @@
 #!/bin/bash
 
+if [ $# -eq 0 ]
+  then
+    echo "Expected one argument: output file"
+    exit 1
+fi
+
 body=$(curl --silent --user-agent "blockscout-solc-bin" "https://api.github.com/repos/vyperlang/vyper/releases" |
```
Comment (leviska) [inline vyper_fetch.sh:9]: does it download all the compilers on run? even when there is no updates? can we make it so it would download only new versions?

Comment (leviska) [reply vyper_fetch.sh:9]: using cache can help, but still, cache can be invalidated and then it will be bad

File: vyper_fetch.sh [hunk 5]
```diff
@@ -1,10 +1,16 @@
 #!/bin/bash
```
Comment (leviska) [inline vyper_fetch.sh:1]: can we move all files related to building vyper into separate folder? like `build_vyper` or something
the list can still be in the root

File: vyper_fetch.sh (hunk 4 above)
Comment (kuksag) [reply vyper_fetch.sh:9]: > does it download all the compilers on run? 

Only the first time, then using cache and wget `--no-clobber` option there will be skips (based on names)

As per [documentation](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) there is no mention of a sudden cache invalidation.
But if we want, I can add some smarter logic

**Comment by kuksag** [comment]
@leviska Thanks for the review.

I made changes according to your comments:
- Made the job run every 3 hours
- Moved related scripts to `vyper` folder
- Made it possible to try downloading only binaries that are absent in our `vyper.list.json`; with that, I removed the "cache" step
It works in a way that it calculates the latest local version from `vyper.list.json` and then fetches only versions that are greater than the found one.
 
Here is the proof of concept: 
1) I removed the 2 latest versions from `vyper.list.json`, on this [branch](https://github.com/kuksag/solc-bin/tree/kuksag/vyper-compiler-ci-test-1) 
2) Successful CI [run](https://github.com/kuksag/solc-bin/runs/7900783325?check_suite_focus=true); also have look at stdout 
3) Here is the following auto-commit: [e8b491f916a384a447a8a301fed9694c0b70302b](https://github.com/kuksag/solc-bin/commit/e8b491f916a384a447a8a301fed9694c0b70302b)

**Review by leviska — APPROVED** [review]

**Pull request merged**